scripts/extensions/pool_update.apply

```diff
@@ -36,16 +36,25 @@ def failure_message(code, params):
     return xmlrpclib.dumps((rpcparams, ), '', True)
 
 
-def execute_apply(session, update_package, yum_conf_file, nogpgcheck):
-    FNULL = open(os.devnull, 'w')
-    retcode = subprocess.call('yum -c ' + yum_conf_file + ' install -y ' +
-                              nogpgcheck + update_package, shell=True, stdout=FNULL, stderr=FNULL)
-    if retcode != 0:
-        raise ApplyFailure(
-            'Failed to install update_package (%r).' % update_package)
+def execute_apply(session, update_package, yum_conf_file):
+    cmd = ['yum', 'install', '-y', '--noplugins', '-c', yum_conf_file, update_package]
+    p = subprocess.Popen(cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, close_fds=True)
+    p.wait()
+    output = p.stdout.read()
+    xcp.logger.info('pool_update.apply %r returncode=%r output=%r', cmd, p.returncode, output)
+    if p.returncode != 0:
+        lines = output.splitlines()
+        errorlines = [l for l in lines if l.startswith('Error: ')]
+        if errorlines:
+            errmsg = errorlines[-1].split(':', 1)[-1].strip()
+        else:
+            errmsg = lines[-1]
```

Comment:
For `yum install` errors, usually the error prefix is 'Error: ', we could confirm `grep 'Error: '` in https://github.com/rpm-software-management/yum
(We didn't add something specific in `yum_conf_file`)